### PR TITLE
Fixed cluster permissions configuration reload

### PR DESCRIPTION
**Pull request by kozlovic**
This is a rework of incorrect changes made in PR #4001. Changes did not work for changes to export permissions. Test was modified to add export changes.

This affects only the `dev` branch.

Signed-off-by: Ivan Kozlovic <ivan@synadia.com>

### CODE REVIEW

File: server/reload.go [hunk 1]
```diff
@@ -2097,64 +2122,53 @@ func (s *Server) reloadClusterPermissions(oldPerms *RoutePermissions) {
 			if canImportNow {
 				// If we could not before, then will need to send a SUB protocol.
 				if !couldImportThen {
-					subsNeedSUB = append(subsNeedSUB, sub)
+					list := subsNeedSUB[route]
```
Comment (derekcollison) [inline server/reload.go:2125]: Just do in one?

```go
list := append(subsNeedSub[route], sub)
```

File: server/reload.go [hunk 2]
```diff
@@ -2097,64 +2122,53 @@ func (s *Server) reloadClusterPermissions(oldPerms *RoutePermissions) {
...
+					list = append(list, sub)
+					subsNeedSUB[route] = list
 				}
 			} else if couldImportThen {
 				// We were previously able to import this sub, but now
 				// we can't so we need to send an UNSUB protocol
-				subsNeedUNSUB = append(subsNeedUNSUB, sub)
+				list := subsNeedUNSUB[route]
```
Comment (derekcollison) [inline server/reload.go:2132]: Same

**Review by derekcollison — COMMENTED** [review]
In general LGTM, small comments.

File: server/reload.go (hunk 1 above)
Comment (kozlovic) [reply server/reload.go:2125]: Or even a one liner with:
```
subsNeedSUB[route] = append(subsNeedSUB[route], sub)
```

**Review by derekcollison — APPROVED** [review]
LGTM

**Pull request merged**